### ISSUE TITLE
Progress this week

### ISSUE THREAD

**Issue by akstoulil** [akstoulil/Group-Project-1#7]
Hello everyone, just checking in on how things for the project are going. I have sent out a few emails previously and not everybody has replied to them. I'm kind of confused as to what everybody is going to be working on this week. I know that Craig has said that he will work on doubling down and the AI a little bit, but other than that no one else has responded with what they are going to work on. Also, for some reason I have some issues with some of the include statements that don't seem to be working as intended. Any feedback is appreciated.

**Comment by minin43** [comment]
Yea I keep procrastinating getting my part done. We really need to figure out how to get it working for Thursday. If anyone has updates they need to push please push them.

**Comment by akstoulil** [comment]
What part are you working on currently by the way? I'd just like to know since I want to make sure that I don't interfere with any of the work you are working on.

**Comment by akstoulil** [comment]
I'm currently working on the main driver some more, and I'm trying to implement some of the player choice/ai code. Also are you having any problems with the external dependencies on your end? There are a lot of ones for me that are giving me errors.

**Comment by minin43** [comment]
What do you mean by external dependencies?

**Comment by akstoulil** [comment]
I guess in my file structure there is a folder for external files. A lot of the include statements seem to link to this folder, and therefore a lot of the dependencies have errors for some reason. Here is a picture of what it looks like.
![image](https://user-images.githubusercontent.com/32655655/36077216-605b4b78-0f2d-11e8-8f98-b7cb551b305c.png)

**Comment by akstoulil** [comment]
Do you have any suggestions as to what we should work on? I'm trying to work on the driver some, but it is a little bit difficult at the current moment, since some of our stuff is incomplete.

**Comment by minin43** [comment]
I'm gonna sit down and get my work done right now and I'll get back to you

**Comment by akstoulil** [comment]
Ok, thanks for letting me know.

**Comment by minin43** [comment]
Addendum: I need the hand class finished first before I can work it into the player class.

**Comment by akstoulil** [comment]
Yeah, I was having some of the same kinds of problems with parts being unfinished. Have you looked at the issue Craig posted on Github about the player class. It contains some code that you could work with. Also I was just wondering if you saw that I pushed an update a few days back that included my updated driver class.

**Comment by akstoulil** [comment]
What are you currently working on again? You mentioned starting on the player class, so did you get finished with something you were working on?

**Comment by minin43** [comment]
I'll take a look at Craig's post.

Yea I saw that and pulled it, so I've got it locally on my machine.

I'm currently working on implementing the player class that we'll use for the player and the opponents

**Comment by CrazyCarl78** [comment]
I pushed the player class. I am going to finish double down. I been caught up with javascript homework so I haven't had a chance to work on it further.

**Comment by minin43** [comment]
Why is the file outside of the program?

Also, it's going to need a lot of revision.

**Comment by CrazyCarl78** [comment]
Sorry, I am still learning to use github.

**Comment by akstoulil** [comment]
It's all good. Thanks for posting the class to Github so we can have access to it! We can revise it and edit it as needed. Also once we get some more parts of the program done, I might need some help with figuring out how to use the pointer object for the deck.

**Comment by CrazyCarl78** [comment]
No problem. I can help with that.

**Comment by CrazyCarl78** [comment]
We could do a structure for the cards with an array in the main program.

**Comment by minin43** [comment]
How to get random cards:
- run through the entire deck array
- if the card class has the variable hasBeenDrawn set to false, copy the card class into a new array
- when you've ran through the entire deck, get 2 random numbers within the bounds of the new array, and you have your 2 cards.
- Mark their respective cards in the master deck hasBeenDraw as true

**Comment by akstoulil** [comment]
Sounds good. I probably need to brush up on pointer notation though. I don't quite remember how to access the array that the SetupDeck function returns.

**Comment by agkingrey** [comment]
Yeah, ive worked on the hand class a bit more. This weekend has been busy with work, but I should get some time tomorrow to hammer out a good portion of it, I'll push out here in the next couple of days.

**Comment by akstoulil** [comment]
Sounds good. I think we should try our best to push out our updates as soon as we can, since the deadline for our prototype is coming up soon. Thanks for the update Austin!

**Comment by agkingrey** [comment]
I think I'm blind haha, where does it say we have to have a working prototype by this week?

**Comment by akstoulil** [comment]
Oh, it doesn't say that on Blackboard. Its just that when Craig and I were in class the last time our teacher had mentioned that we should try to get a working prototype by the next class period. It doesn't have to be a finished product, but it should be something that runs.

**Comment by agkingrey** [comment]
Gotcha, I sent him an email asking for anything I missed on thurs., he must have forgotten to put that in there.

**Comment by akstoulil** [comment]
Also, is there any particular way I should store the drawn cards? I'm not sure if you are going to have an array or vector for them in the hand class or not, so I'm not sure if I should store them in an array/vector in the main program, or if I should wait for the update on the hand class.

**Comment by minin43** [comment]
I'd imagine we'll store them in the hand class, same with additional draws as well as the hand's bet.

**Comment by minin43** [comment]
We have to allow for the reaction of infinite new hands and infinite (although impossible) draws per hand.

**Comment by akstoulil** [comment]
Ok. I'll have to make some revisions to my code once the update gets pushed. I think we might just create an array/vector in the hand class, and store all of the cards we get from the deck there. Do you know how to access the array pointer that the SetupDeck function by the way? I've been having trouble recently with getting that to work. Also we could create a vector in the player class to store the number of hands the player has.

**Comment by minin43** [comment]
> I think we might just create an array/vector in the hand class, and store all of the cards we get from the deck there

We're going to have to store that information dynamically, which means more pointer notation. You can't create an array of infinite size, we have to increase and decrease its size manually. I'm not sure how to do that, I was expecting to have to look it up myself.

> Do you know how to access the array pointer that the SetupDeck function by the way?
I don't think there's anything special needed. Just DeckObject should work, or it not, *DeckObject.

**Comment by akstoulil** [comment]
The thing with creating an array of infinite size could be possible with the use of vectors, since vectors can expand for every item added, and can also have said items removed.

**Comment by minin43** [comment]
I'm not versed in vector usage.

**Comment by akstoulil** [comment]
That's ok, vectors are fairly straightforward. I would mainly suggest using them because they can expand and contract on the fly, and you can traverse them in a way that is very similar to an array. Would you be for or against the usage of them? I feel that they could make the whole thing of arrays that need their size changed much easier.

**Comment by agkingrey** [comment]
Okay, please let me know what we should do as a solution going forward. Currently, we have our setupdeck function returning a pointer to the array created in the function.  The problem is, is that the pointer created, actually loses all of the data in the array, once it leaves the class, and is simply a pointer. This makes it impossible to pull or edit from the deck object created, because the value I am able to grab is a pointer to an empty array.  Here's some way smarter people explaining it better:

https://stackoverflow.com/questions/31194273/passing-an-array-into-a-class-function-from-another-classs-constructor

**Comment by agkingrey** [comment]
****Please read before pulling what I pushed tonight*****
I went ahead and pushed what I got done so far. There will be some errors until we address what I posted above.  The parameters that my addcard and drawtwo functions are using are just "placeholders" until we can convert our deck to a vector, or figure out another solution. I wrote the code to be able to easily switch it over once we fix.

**Comment by akstoulil** [comment]
Hello everyone. Sorry about not updating you all with what I have been working on. Currently I have been working on making various revisions to the project to get it into a working state. I currently have a project that runs, and lets the user choose from various options. I am wondering what you guys think of it so far. By the time you read this comment, there should be another branch on the project that contains the project that I have been working on. Also, just to answer some questions about the project I have been working on, the project solution and structure is different from the one that we currently have on Github because I have been getting a lot of very strange errors from the repository that I have downloaded. These errors don't seem to be affecting anyone else, so I just created this new project in hopes of getting one that works for everyone. Also the code for the hand class has been commented out in my program currently due to some errors being generated, but I will try to look further into them. Any feedback on this is appreciated. Sorry again for not updating you guys sooner.

**Comment by CrazyCarl78** [comment]
Looks good. Thanks for the update.

**Comment by akstoulil** [comment]
Hey guys, just checking in on how things are going. I was just wondering if everyone has had a chance to look over what I have so far. Also what are we going to do for the prototype tomorrow? I currently have a working prototype on the test branch that I have created, so are we going to want to use that?

**Comment by agkingrey** [comment]
We probably should use that. I'm still waiting on everyone's answer to the question I asked above, so we can have a working deck and hand.

**Comment by akstoulil** [comment]
Sorry about not answering the question. What are you needing to ask about? Is it about converting our deck to a different datatype? Because in my prototype I believe that I just used a vector in the player class to store our cards in, and used a 2D array to store the deck of cards to choose from.

**Comment by minin43** [comment]
Sorry for not responding the past few days, been really busy with classes and life stuff (car accident shit) so it's really hard to find that motivated groove to do some coding.

I'll try and take a look at everyone's code tomorrow and provide some feedback over what we decide the working prototype should look like, and we can use our self-evaluation to continue working on the project.
